src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java

```diff
@@ -64,4 +65,21 @@ public RedirectView redirectAfterColaLogin(final @CookieValue(name = REDIRECT_UR
 
         return new RedirectView(redirectUrl.orElse(configProperties.getDefaultRedirectUrl()));
     }
+
+    @GetMapping("/refresh-token")
+    public ResponseEntity<String> refreshToken(@CookieValue(USER_SERVICE_COOKIE_NAME) final String currentToken, final HttpServletResponse response) {
+
+        // presumably the security filter will handle invalid/non-existent tokens and deny access but if not then add code below.
+
+        customJwtService.addTokenToBlacklist(currentToken);
+
+        final String newToken = customJwtService.generateToken();
+        final Cookie userTokenCookie = new Cookie(USER_SERVICE_COOKIE_NAME, newToken);
+        userTokenCookie.setSecure(true);
+        userTokenCookie.setHttpOnly(true);
```

Comment:
Wonder if its worth making a little util that makes a cookie with secure & http only by default? Also wondering if we would want a domain to be set for these cookies, or the strict mode